File: Activity2.1/app/src/main/java/com/mobiledevices/miguel/activity21/JSONAdapter.java
package com.mobiledevices.miguel.activity21;

import android.app.Activity;
import android.support.v4.app.ActivityCompat;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by miguel on 21/03/17.
 */

public class JSONAdapter extends BaseAdapter {

    private JSONArray data;
    private Activity activity;

    public JSONAdapter(Activity a, JSONArray d) {
        this.data = d;
        this.activity = a;
    }

    @Override
    public int getCount() {
        return this.data.length();
    }

    @Override
    public Object getItem(int position) {
        try {

            return data.getJSONObject(position);
        }
        catch(JSONException jse) {

            jse.printStackTrace();
        }

        return null;
    }

    @Override
    public long getItemId(int position) {
        return this.data.length();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if(convertView == null) {
            convertView = this.activity.getLayoutInflater().inflate(R.layout.row_friends, null);
        }

        TextView name = (TextView) convertView.findViewById(R.id.row_name);
        TextView hobby = (TextView) convertView.findViewById(R.id.row_hobby);

        try {
            JSONObject current = this.data.getJSONObject(position);
            name.setText(current.getString("name"));
            hobby.setText(current.getString("hobby"));
        }
        catch(JSONException e) {
            e.printStackTrace();
        }

        return convertView;

    }
}

File: Activity3.3/Activity_3.3/JSONAdapter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Org.Json;
using Android.Util;

namespace Activity_3._3
{
    class JSONAdapter : BaseAdapter
    {
        Activity activity;
        JSONArray data;

        public JSONAdapter(Activity activity, JSONArray data)
        {
            this.activity = activity;
            this.data = data;
        }

        public override int Count => data.Length();

        public override Java.Lang.Object GetItem(int position)
        {
            return data.GetJSONObject(position);
        }

        public override long GetItemId(int position)
        {
            return data.GetJSONObject(position).GetInt("id");
        }

        public override View GetView(int position, View convertView, ViewGroup parent)
        {
            if (convertView == null)
            {
                convertView = LayoutInflater.From(this.activity).Inflate(Resource.Layout.JSON_row, null);
            }

            TextView rowTitle = convertView.FindViewById<TextView>(Resource.Id.tv_row_title);
            TextView rowBody = convertView.FindViewById<TextView>(Resource.Id.tv_row_body);
            JSONObject post = (JSONObject)this.GetItem(position);

            rowTitle.Text = post.GetString("title");
            rowBody.Text = post.GetString("body");

            return convertView;
        }
    }
}